src/compiler/factoryPublic.ts

```diff
@@ -218,6 +218,9 @@ namespace ts {
 
     // Private Identifiers
     export function createPrivateIdentifier(text: string): PrivateIdentifier {
+        if (text[0] !== "#") {

```

Comment:
I guess it's not too bad to omit the `length` check.

Reply:
```
$ node
Welcome to Node.js v12.13.1.
Type ".help" for more information.
> ""[0]
undefined
> ""[0] !== "#"
true
```

We :heart: JavaScript